crates/cadenza/src/mcp.rs

```diff
@@ -0,0 +1,430 @@
+    #[tool(description = "List all built-in functions, operators, and special forms available in Cadenza. Use this to discover available functionality.")]
+    async fn list_builtins(&self) -> Result<CallToolResult, McpError> {
+        let builtins = [
+            "Operators: +, -, *, /, ==, !=, <, <=, >, >=, |>",
+            "Special Forms: let, fn, match, assert, typeof, measure",
+            "Math Functions: (to be implemented)",
+            "List Functions: (to be implemented)",
+            "String Functions: (to be implemented)",
+        ];
+
+        Ok(CallToolResult::success(vec![Content::text(
+            builtins.join("\n"),
+        )]))
+    }
+
+    /// Get documentation for a symbol
+    #[tool(description = "Get documentation for a specific symbol, function, or operator. Use this to understand how to use language features.")]
+    async fn get_docs(
+        &self,
+        rmcp::handler::server::wrapper::Parameters(req): rmcp::handler::server::wrapper::Parameters<
+            GetDocsRequest,
+        >,
+    ) -> Result<CallToolResult, McpError> {
+        let docs = match req.symbol.as_str() {
+            "let" => "let name = expression\n\nBind a value to a name in the current scope.",
+            "fn" => "fn name param1 param2 ... = body\n\nDefine a function with parameters.",
+            "measure" => {
+                "measure name\nmeasure name = base_unit factor\n\nDefine a unit of measurement."
+            }
+            "+" => "a + b\n\nAddition operator. Works with numbers and compatible units.",
+            "-" => "a - b\n\nSubtraction operator. Works with numbers and compatible units.",
+            "*" => "a * b\n\nMultiplication operator. Can multiply values with units.",
+            "/" => "a / b\n\nDivision operator. Can divide values with units.",
+            "|>" => "a |> f\n\nPipeline operator. Passes the left value as argument to the right function.",
+            _ => "No documentation available for this symbol.",
+        };
+
+        Ok(CallToolResult::success(vec![Content::text(docs)]))
+    }
+
+    /// Get information about the Cadenza language
+    #[tool(description = "Get comprehensive information about the Cadenza language, its features, and use cases.")]
+    async fn about_cadenza(&self) -> Result<CallToolResult, McpError> {
+        let about = r#"# Cadenza Programming Language
+
+Cadenza is a functional programming language with first-class support for units of measure and dimensional analysis.
+
+## Key Features
+
+1. **Units of Measure**: Built-in support for physical units and automatic dimensional analysis
+   - Define base units: `measure meter`
+   - Define derived units: `measure kilometer = meter 1000`
+   - Use in expressions: `10meter`, `5kilometer`
+   - Automatic unit checking: prevents adding incompatible units
+
+2. **Functional Programming**: Functions are first-class values
+   - Define functions: `fn square x = x * x`
+   - Anonymous functions: `fn x -> x + 1`
```

Comment:
Anonymous functions haven't been implemented and we don't have a syntax for them yet.